disco_aws_automation/disco_vpc.py

```diff
@@ -641,307 +429,63 @@ def _destroy_rds(self, wait=True):
-
-    def get_all_security_groups_for_vpc(self):
-        """ Find all security groups belonging to vpc and return them """
-        return self.vpc.connection.get_all_security_groups(filters=self.vpc_filter())
-
-    def _destroy_igws(self):
-        """ Find all gateways belonging to vpc and destroy them"""
-        vpc_attachment_filter = {"attachment.vpc-id": self.vpc.id}
-        # delete gateways
-        for igw in self.vpc.connection.get_all_internet_gateways(filters=vpc_attachment_filter):
-            self.vpc.connection.detach_internet_gateway(igw.id, self.vpc.id)
-            self.vpc.connection.delete_internet_gateway(igw.id)
+        for subnet in self.boto3_ec2.describe_subnets(Filters=self.vpc_filters())['Subnets']:
+            self.boto3_ec2.delete_subnet(SubnetId=subnet['SubnetId'])
 
     def _destroy_routes(self):
         """ Find all route_tables belonging to vpc and destroy them"""
-        for route_table in self.vpc.connection.get_all_route_tables(filters=self.vpc_filter()):
-            if not route_table.tags:
-                logging.info("Skipping untagged (default) route table %s", route_table.id)
+        for route_table in self.boto3_ec2.describe_route_tables(Filters=self.vpc_filters())['RouteTables']:
+            if len(route_table["Associations"]) > 0 and route_table["Associations"][0]["Main"]:
+                logging.info("Skipping the default main route table %s", route_table['RouteTableId'])
                 continue
             try:
-                self.vpc.connection.delete_route_table(route_table.id)
+                self.boto3_ec2.delete_route_table(RouteTableId=route_table['RouteTableId'])
             except EC2ResponseError:
-                logging.error("Error deleting route_table %s:.", route_table.id)
+                logging.error("Error deleting route_table %s:.", route_table['RouteTableId'])
                 raise
 
     def _destroy_vpc(self):
         """Delete VPC and then delete the dhcp_options that were associated with it. """
 
-        # save function and parameters so we can delete dhcp_options after vpc. We do this becase botos
-        # get_all_dhcp_options does not support filter. Because we cannot easily find the default dhcp
-        # options, re-assigning default dhcp option is not trivial.
-        delete_dhcp_options = self.vpc.connection.delete_dhcp_options
-        dhcp_options_id = self.vpc.dhcp_options_id
+        # save function and parameters so we can delete dhcp_options after vpc.
+        dhcp_options_id = self.vpc['DhcpOptionsId']
 
-        delete_status = keep_trying(30, self.vpc.delete)
+        self.boto3_ec2.delete_vpc(VpcId=self.get_vpc_id())
         self.vpc = None
 
-        if not delete_dhcp_options(dhcp_options_id):
-            logging.warning("failed to delete dhcp options (%s)", dhcp_options_id)
-
-        return delete_status
+        self.boto3_ec2.delete_dhcp_options(DhcpOptionsId=dhcp_options_id)
```

Comment:
We could try to delete only the DCHP options set that we created, which is tagged with name of the VPC.